src/NumField/NfAbs/NonSimple.jl

```diff
@@ -1058,7 +1058,7 @@ function minpoly_via_trace(a::AbsNonSimpleNumFieldElem)
       i += 1
     end
     q = QQFieldElem(1, div(d, i))
-    f = power_sums_to_polynomial([x*q for x = l])
+    f = power_sums_to_polynomial([x*q for x = l]; parent = Qx)
     if iszero(subst(f, a))  #TODO: to checks first...
       return f::QQPolyRingElem
```

Comment:
Ah so there is a type assertion here, too -- so the outer one should not be necessary?

But if we need that, it means `power_sums_to_polynomial` is type unstable, and we really should do
```suggestion
    f = power_sums_to_polynomial([x*q for x = l]; parent = Qx)::QQPolyRingElem
    if iszero(subst(f, a))  #TODO: to checks first...
      return f
```
so that we also benefit in the `subst` call.